projects/laji/src/app/+taxonomy/taxon/info-card/taxon-identification/taxon-identification.component.ts

```diff
@@ -1,137 +1,50 @@
-import { ChangeDetectionStrategy, Component, Input, OnInit, OnChanges, SimpleChanges, ChangeDetectorRef } from '@angular/core';
+import { ChangeDetectionStrategy, Component, Input, OnInit, OnChanges, SimpleChanges, ChangeDetectorRef, OnDestroy } from '@angular/core';
 import { Taxonomy } from '../../../../shared/model/Taxonomy';
 import { Image } from '../../../../shared/gallery/image-gallery/image.interface';
-import { switchMap, map } from 'rxjs/operators';
-import { forkJoin, Observable, of } from 'rxjs';
-import { TranslateService } from '@ngx-translate/core';
-import { PagedResult } from 'projects/laji/src/app/shared/model/PagedResult';
-import { TaxonomyApi } from 'projects/laji/src/app/shared/api/TaxonomyApi';
-
-type TaxonChildren = {
-  taxonomy: Taxonomy,
-  species: PagedResult<Taxonomy>
-}[];
-
-interface TaxonomyWithChildren extends Taxonomy {
-  children: TaxonomyWithChildren[];
-}
-
-const rankWhiteList = [
-  'MX.superdomain',
-  'MX.domain',
-  'MX.kingdom',
-  'MX.phylum',
-  'MX.class',
-  'MX.order',
-  'MX.family',
-  'MX.genus',
-  'MX.species'
-];
+import { TaxonIdentificationFacade } from './taxon-identification.facade';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'laji-taxon-identification',
   templateUrl: './taxon-identification.component.html',
   styleUrls: ['./taxon-identification.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class TaxonIdentificationComponent implements OnInit, OnChanges {
+export class TaxonIdentificationComponent implements OnInit, OnChanges, OnDestroy {
+  private unsubscribe$ = new Subject<void>();
+
   @Input() taxon: Taxonomy;
   @Input() taxonImages: Array<Image>;
 
-  taxonChildren: TaxonChildren = [];
+  childTree: Taxonomy[] = [];
 
   loading = false;
 
-  constructor(private taxonomyApi: TaxonomyApi, private cdr: ChangeDetectorRef, private translate: TranslateService) { }
+  constructor(private facade: TaxonIdentificationFacade, private cdr: ChangeDetectorRef) { }
 
-  ngOnInit() {}
+  ngOnInit() {
+    this.facade.childTree$.pipe(takeUntil(this.unsubscribe$)).subscribe(d => {
```

Comment:
you could leverage async pipe to remove cdr requirement all together. This would be 

`childTree$: Observable<Taxonomy[]> = of([]);`

move this to ngOnChanges and do something like this:
```
let childTree$;

if (this.taxon.taxonRank === 'MX.species') {
  childTree$ = of([]);
} else {
  childTree$ = this.facade.childTree$;
  this.facade.loadChildTree(this.taxon);
}

this.loading = true;
this.childTree$ = tree$.pipe(
  tap(() => this.loading = false) // this loading could be removed since I didn't see it in the view (with a quick look :))
)
```
You could then remove the takeUntil, ChangeDetectorRef, ngOnInit, ngOnDestroy and unsubscribe$ all together